client/lib/sites-list/list.js

```diff
@@ -394,7 +394,7 @@ SitesList.prototype.setRecentlySelectedSite = function( siteID ) {
 };
 
 SitesList.prototype.getRecentlySelected = function() {
-	if ( ! this.recentlySelected.length ) {
+	if ( ! this.recentlySelected.length || ! this.initialized ) {
```

Comment:
This is a good case of relying on different stores and how that becomes difficult in these old data structures.